CIP-0015/schema.cddl

```diff
@@ -8,12 +8,16 @@ $staking_pub_key /= bytes .size 32
 $ed25519_signature /= bytes .size 64
 $reward_address /= bytes
 $nonce /= uint
+$prop /= uint
+$voting_purpose /= uint
+delegation = (vote_key: $voting_pub_key, proportion: $prop)
 
 key_registration = {
-  1 : $voting_pub_key,
+  1 : [+delegation],
```

Comment:
presumably this should be backwards compatible?

Reply:
It could be useful indeed, but that means we should also have a default implicit value for `voting_purpose` (which could be catalyst)

Reply:
If it's not mean to be backwards compatible, we could probably specify in the CIP a cutoff point where all registrations using the old format are no longer valid. If we don't want this kind of cutoff point, then the CIP should maintain backwards compatibility

Reply:
Added a legacy version for the `voting_key` field and used 0 as the default for `voting_purpose` since old registrations were meant for Catalyst only

Reply:
Yes, going forwards, we will have many voting purposes (both funding and governance).  As you say, setting the default to be 0 (and assigning this to Catalyst) seems sensible.

Reply:
Concerning invalidating existing registrations, we will need to see how the Catalyst team use the new capability.  They may well be happy with (or positively want) retaining existing delegations.   New uses will need to use the new format, of course (we definitely want to allow different delegation choices for different voting purposes).  There will be some semantics associated with the purpose, but that is essentially assigned externally to the chain.  It would be possible to use the same purpose for multiple voting events if wanted.  We shouldn't mandate that semantics.

As I see it, the advantage of invalidating old registrations would be

1. all voters will need to think about their allocations (and not just continue with a single delegation)
2. front-end and bridge tooling only needs to support one CIP-15 format

The disadvantages would be:

1. voter disruption (need to re-register)
2. cost (for each voter submitting a new registration)